src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleFiltering/components/LocationFilter/index.tsx

```diff
@@ -83,7 +83,9 @@ export const LocationFilter = ({
         namedLocations.find(
           (location) => location.name === newSelectedOptions.name
         ) || null;
-      setSelectedLocation(newLocation);
+      if (newLocation && newLocation.id !== selectedLocation?.id) {
+        setSelectedLocation(newLocation);
+      }
```

Comment:
We need to avoid calling setSelectedLocation if there is no change because if we make the extra call the parent will think that a filter is enabled and the reset button will show when it's not supposed to.